keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/AgpVersionHandler.kt

```diff
@@ -1,107 +0,0 @@
-/*
- * Copyright (C) 2020 Slack Technologies, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    https://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-@file:Suppress("deprecation")
-package com.slack.keeper
-
-import com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
-import com.android.build.gradle.internal.tasks.ProguardConfigurableTask
-import com.google.auto.service.AutoService
-import org.gradle.api.Project
-import org.gradle.api.file.Directory
-import org.gradle.api.provider.Provider
-import org.gradle.kotlin.dsl.withType
-import org.gradle.util.VersionNumber
-import java.util.Locale
-
-/** A handler API for working with different AGP versions. */
-// TODO(zsweers) could we make this Serializable for a task input?
-interface AgpVersionHandler {
-
-  companion object {
-    fun getInstance(): AgpVersionHandler = lazyInstance
-
-    private val lazyInstance by lazy {
-      createHandler()
-    }
-
-    /** Creates a new patcher for the given environment. */
-    private fun createHandler(): AgpVersionHandler {
```

Comment:
Should we keep some of this scaffolding around since we may reuse it for future AGP versions? Or is it easy enough to rewrite? 🤷

Reply:
Easy enough to rewrite!